### PR TITLE
Feature/um metrics 01

### PR DESCRIPTION
**Pull request by fivsag**

### CODE REVIEW

**Review by miunsag — CHANGES_REQUESTED** [review]
Please produce a README file in the same folder as the docker-compose.yml file. Each project will have their own README.
Second, I have mapped a valid license file but it went in error. It seems it wants to mount a folder (?)

in .env:
UM_LICENCE_LOCATION=c:\v\Lic\exp_2024-11-30\wm10x-current\Linux\Universal Messaging Integr AP\0000763146_UniversalMessaging_Integration105.xml

error:
Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/mnt/c/v/Lic/exp_2024-11-30/wm10x-current/Linux/Universal Messaging Integr AP/0000763146_UniversalMessaging_Integration105.xml" to rootfs at "/opt/softwareag/UniversalMessaging/server/umserver/licence": mount /mnt/c/v/Lic/exp_2024-11-30/wm10x-current/Linux/Universal Messaging Integr AP/0000763146_UniversalMessaging_Integration105.xml:/opt/softwareag/UniversalMessaging/server/umserver/licence (via /proc/self/fd/6), flags: 0x5000: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

File: compose/um-metrics-02/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+networks:
+  n1:
+    external: false
+
+services:
+  umnode:
+    image: ${UM_CONTAINER_IMAGE_NAME}
+    ports:                         
+     - 9000:9000
+     - 9200:9200
```
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
      - 9000:9000
      - 9200:9200
```

Comment (miunsag) [reply compose/um-metrics-02/docker-compose.yml]: Indentation is missing a space?

File: compose/um-metrics-02/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    volumes:
+     - um_data:/opt/softwareag/UniversalMessaging/server/umserver/data
+     - um_logs:/opt/softwareag/UniversalMessaging/server/umserver/logs
+     - ${UM_LICENCE_LOCATION}:/opt/softwareag/UniversalMessaging/server/umserver/licence
+     - um_users:/opt/softwareag/common/conf
```
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
      - um_data:/opt/softwareag/UniversalMessaging/server/umserver/data
      - um_logs:/opt/softwareag/UniversalMessaging/server/umserver/logs
      - ${UM_LICENCE_LOCATION}:/opt/softwareag/UniversalMessaging/server/umserver/licence
      - um_users:/opt/softwareag/common/conf
```

Comment (miunsag) [reply compose/um-metrics-02/docker-compose.yml]: Indentation is missing a space?

File: compose/um-metrics-02/helpers/cleaner.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+# remove all images
+# echo '####################################################'
+# echo 'Removing images ...'
+# echo '####################################################'
+# docker rmi $(docker images -q)
+
+# remove all stray volumes if any
+# echo '####################################################'
+# echo 'Revoming docker container volumes (if any)'
```
Comment (miunsag) [inline compose/um-metrics-02/helpers/cleaner.sh]:
```suggestion
# echo 'Removing docker container volumes (if any)'
```

File: compose/um-metrics-02/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+services:
+  umnode:
+    image: ${UM_CONTAINER_IMAGE_NAME}
+    ports:                         
+     - 9000:9000
+     - 9200:9200
+    volumes:
+     - um_data:/opt/softwareag/UniversalMessaging/server/umserver/data
+     - um_logs:/opt/softwareag/UniversalMessaging/server/umserver/logs
+     - ${UM_LICENCE_LOCATION}:/opt/softwareag/UniversalMessaging/server/umserver/licence
```
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
     - ${UM_LICENCE_LOCATION}:/opt/softwareag/UniversalMessaging/server/umserver/licence/
```

Comment (miunsag) [reply compose/um-metrics-02/docker-compose.yml]: Add a slash to quickly tell the yaml reader this is a directory mount, not a file mount.
Besides, add a note in the README that the directory needs to contain a file called exactly "licence.xml"

File: compose/um-metrics-02/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+     - um_users:/opt/softwareag/common/conf
+    networks:
+      n1:
+        aliases:
+          - "um-node"
+
+  prometheus:
+    image: prom/prometheus
+    ports:
+      - 9090:9090
```
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
      - ${H_SUIF_PORT_PREFIX}90:9090
```

File: compose/um-metrics-02/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+networks:
+  n1:
+    external: false
+
+services:
+  umnode:
+    image: ${UM_CONTAINER_IMAGE_NAME}
+    ports:                         
+     - 9000:9000
```
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
     - ${H_SUIF_PORT_PREFIX}99:9000
```

File: compose/um-metrics-02/docker-compose.yml (hunk 1 above)
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
     - ${H_SUIF_PORT_PREFIX}92:9200
```

File: compose/um-metrics-02/docker-compose.yml (hunk 5 above)
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
      - ${H_SUIF_PORT_PREFIX}99:9090
```

File: compose/um-metrics-02/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+      - /proc:/host/proc:ro
+      - /sys:/host/sys:ro
+      - /:/rootfs:ro
+    command:
+      - '--path.procfs=/host/proc'
+      - '--path.rootfs=/rootfs'
+      - '--path.sysfs=/host/sys'
+      - '--collector.filesystem.mount-points-exclude=^/(sys|proc|dev|host|etc)($$|/)'
+    ports:
+      - 9100:9100
```
Comment (miunsag) [inline compose/um-metrics-02/docker-compose.yml]:
```suggestion
      - ${H_SUIF_PORT_PREFIX}91:9100
```

Comment (miunsag) [reply compose/um-metrics-02/docker-compose.yml]: Actually the besides the Grafana port, and maybe UM, any other port may be not mapped at all...

**Comment by fivsag** [comment]
- README file moved
- Please use a relative path to the local folder that contains the valid UM license file. eg:
image

**Review by miunsag — CHANGES_REQUESTED** [review]
Please observe the comments and suggestions

**Review by miunsag — APPROVED** [review]

**Pull request merged**